service/src/main/java/io/camunda/service/DecisionRequirementsServices.java

```diff
@@ -48,7 +50,16 @@ public DecisionRequirementsServices withAuthentication(final Authentication auth
   public SearchQueryResult<DecisionRequirementsEntity> search(
       final DecisionRequirementsQuery query) {
     return decisionRequirementSearchClient.searchDecisionRequirements(
-        query, SecurityContext.of(s -> s.withAuthentication(authentication)));
+        query,
+        SecurityContext.of(
```

Comment:
💭  [Suggestion/Optional]  I see this is used over all the services, and it is a bit long, would be the case to extract this to a utility class and reuse? 

for example: 
```
public class SecurityContextUtil {

    public static SecurityContext buildSecurityContext(Authentication authentication,
                                                      SecurityConfiguration securityConfig,
                                                      AuthorizationResourceType resourceType,
                                                      PermissionType permissionType) {
        return SecurityContext.of(
            s -> s.withAuthentication(authentication)
                  .withAuthorizationIfEnabled(
                      securityConfig.getAuthorizations().isEnabled(),
                      a -> a.resourceType(resourceType)
                            .permissionType(permissionType)
                  )
        );
    }
}
```

Dont need to be done in this PR, its maybe an idea.

Reply:
I thought of doing that too, I will see if I can do this when implementing permission checks on the Get by key endpoints